File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1

    l = []
    for i in s :
        if l :  
            if l[-1] == i :
                l.pop()
            else :
                l.append(i)
        else :
            l.append(i)
    
    if l : return 0
    else : return 1
    

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution {
public static int solution(String s) {
        ArrayDeque<Character> stack = new ArrayDeque<>();
        for (char c : s.toCharArray()) {
            if (stack.isEmpty()) {
                stack.addLast(c);
            } else {
                if (c == stack.peekLast()) {
                    stack.pollLast();
                } else {
                    stack.addLast(c);
                }
            }
        }

        if (stack.isEmpty()) {
            return 1;
        }
        return 0;
    }
}